multiview/cluster/mv_k_means.py

```diff
@@ -0,0 +1,330 @@
+from scipy.spatial.distance import cdist
+
+
+class MultiviewKMeans(BaseCluster):
+
+    '''
+    An implementation of Multi-View K-Means using the co-EM algorithm.
+    This algorithm currently handles two views of data.
+
+    Parameters
+    ---------
+    n_clusters : int, optional, default=2
+        The number of clusters
+
+    random_state : int, optional, default=None
+        Determines random number generation for initializing centroids.
+        Can seed the random number generator with an int.
+
+    patience : int, optional, default=5
+        The number of EM iterations with no decrease in the objective
+        function after which the algorithm will terminate.
+
+    max_iter : int, optional, default=None
+        The maximum number of EM iterations to run before
+        termination.
+
+    n_init : int, optional, default=5
+        Number of times the k-means algorithm will run on different
+        centroid seeds. The final result will be the best output of
+        n_init runs with respect to total inertia across all views.
+
+    Attributes
+    ----------
+
+    centroids_ : list of array-likes
+        - centroids_ length: n_views
+        - centroids_[i] shape: (n_clusters, n_features_i)
+        The cluster centroids for each of the two views. centroids_[0]
+        corresponds to the centroids of view 1 and centroids_[1] corresponds
+        to the centroids of view 2.
+
+    References
+    ----------
+    [1] Bickel S, Scheffer T (2004) Multi-view clustering. Proceedings of the
+    4th IEEE International Conference on Data Mining, pp. 19–26
+    '''
+
+    def __init__(self, n_clusters=2, random_state=None,
+                 patience=5, max_iter=None, n_init=5):
+
+        super().__init__()
+
+        if not (isinstance(n_clusters, int) and n_clusters > 0):
+            msg = 'n_clusters must be a positive integer'
+            raise ValueError(msg)
+
+        if random_state is not None:
+            msg = 'random_state must be convertible to 32 bit unsigned integer'
+            try:
+                random_state = int(random_state)
+            except ValueError:
+                raise ValueError(msg)
+            np.random.seed(random_state)
+
+        if not (isinstance(patience, int) and (patience > 0)):
+            msg = 'patience must be a nonnegative integer'
+            raise ValueError(msg)
+
+        self.max_iter = None
+        if max_iter is not None:
+            if not (isinstance(max_iter, int) and (max_iter > 0)):
+                msg = 'max_iter must be a positive integer'
+                raise ValueError(msg)
+            self.max_iter = max_iter
+
+        if not (isinstance(n_init, int) and (n_init > 0)):
+            msg = 'n_init must be a nonnegative integer'
+            raise ValueError(msg)
+
+        self.n_clusters = n_clusters
+        self.random_state = random_state
+        self.patience = patience
+        self.n_init = n_init
+        self.centroids_ = None
```

Comment:
don't need this I don't think

Reply:
Do you mean the centroids attribute? I think the centroids need to be saved as an attribute during training in order to use them during predictions.